library/aes.c

```diff
@@ -397,8 +397,14 @@ static int aes_init_done = 0;
 static void aes_gen_tables( void )
 {
     int i, x, y, z;
-    int pow[256];
-    int log[256];
+#if !defined(MBEDTLS_AES_FEWER_TABLES)
+    uint32_t *pow = RT2;
```

Comment:
will not work on all systems. `sizeof(int)` is not necessary 32 bits.
Although `i` and `x` will not exceed 255, there could be at least compilation warnings, when `int` is not 32 bits

Reply:
Ok you are right but we should keep `int` because of [Integer promotion](https://wiki.sei.cmu.edu/confluence/display/c/INT02-C.+Understand+integer+conversion+rules)

Result of such operation
```
#define ROTL8(x) ( ( x << 8 ) & 0xFFFFFFFF ) | ( x >> 24 )
```

would be an `int` result. Yeah we would have problem here if we use not 32 bit int

So maybe `i, x, y, z` should be `uint32_t` ?